src/frontend/screens/AuthWrapper.tsx

```diff
@@ -0,0 +1,17 @@
+import * as React from "react";
+import { createAppContainer } from "react-navigation";
+import AppContainerWrapper from "../AppContainerWrapper";
+import { SecurityContext } from "../context/SecurityContext";
+import { AuthStack } from "../Navigation/AuthStack";
+
+const AuthContainer = createAppContainer(AuthStack);
+
+export const AuthWrapper = () => {
```

Comment:
think the conditional rendering here is causing a moment where the screen is completely grey for about a second after successful passcode input. could the authscreen be implemented as a modal to avoid this issue? would have to play around with it otherwise to figure out how to fix

Reply:
The problem with the modal, is android has no real way to tracking when it gets closed, so we are unable to add the modal to navigation state on close. So when the user closes the app, and opens again, the nav state persists to NOT the auth screen. The initial loading screen/app screen doesn have access to the navigation, so there is not way of dispatching the modal in that loading screen

Reply:
makes sense. i think that brings up another question of if it's even necessary for this screen to be part of the navigation stack. unless there's something about deeplinking or something along those lines, it doesn't seem like we need to create a new stack navigator for this auth screen. the auth screen appearing is essentially a function of the security context state, and not sure if it'll ever need to be part of the navigation stack to accomplish its purpose. for example, i think if you did something like:

```tsx
// AuthWrapper.tsx
import * as React from "react";
import { createAppContainer } from "react-navigation";
import AppContainerWrapper from "../AppContainerWrapper";
import { SecurityContext } from "../context/SecurityContext";
// import { AuthStack } from "../Navigation/AuthStack";
import { AuthScreen } from "./AuthScreen";

// const AuthContainer = createAppContainer(AuthStack);

export const AuthWrapper = () => {
  const [{ authStatus }] = React.useContext(SecurityContext);

  if (authStatus === "pending") {
    // return <AuthContainer />;
    return <AuthScreen />;
  }

  return <AppContainerWrapper />;
};
```

with the accompanying style change in `AuthScreen.tsx`:

```tsx
  container: {
    paddingHorizontal: 20,
    alignItems: "center",
    // justifyContent: "center",
    // marginTop: 40,
    paddingTop: 40,
    backgroundColor: WHITE,
    flex: 1,
  },
```
you could accomplish the same thing without dealing with navigation complexity. let me know if there's something im missing though

Reply:
I changed this to a modal and had to add an intermediate loading screen!